### PR TITLE
Math Magicians : Events

### PR DESCRIPTION
**Pull request by ohyajk**
I added Operate.js and calculate.js in a Logic folder to add events to the Math Magician project.

### CODE REVIEW

File: src/components/calculator.jsx [hunk 1]
```diff
@@ -1,42 +1,62 @@
-import React, { Component } from 'react';
+import { Component } from 'react';
+import calculate from '../logic/calculate';
 // eslint-disable-next-line react/prefer-stateless-function
-export default class calculator extends Component {
+class calculator extends Component {
```
Comment (TedLivist) [inline src/components/calculator.jsx:4]: - [x] Consider displaying the number pressed on clicking it. Right now if I click a number it does not appear on the screen until I click an operator button. Kindly ensure the numbers clicked are displayed on the calculator without having to wait for an operator click.

  For example, if I want to perform this operation, "89 x 5", "89" does not appear until I click "x", and 5 is not displayed at all - it just displays the final result, "445"

File: src/components/calculator.jsx [hunk 2]
```diff
@@ -1,42 +1,62 @@
...
+            <li className="flex justify-center border-2 border-black/40 w-1/4"><button className="w-full" type="button" buttonName="%" data-id="%" onClick={this.clickHandler.bind(this)}>%</button></li>
+            <li className="flex justify-center border-2 border-black/40 w-1/4"><button className="w-full" type="button" buttonName="÷" data-id="÷" onClick={this.clickHandler.bind(this)}>÷</button></li>
           </ul>
           <ul className="flex justify-center items-center">
-            <li className="flex justify-center border-2 border-black/40 w-1/4">7</li>
-            <li className="flex justify-center border-2 border-black/40 w-1/4">8</li>
-            <li className="flex justify-center border-2 border-black/40 w-1/4">9</li>
-            <li className="flex justify-center border-2 border-black/40 w-1/4">x</li>
+            <li className="flex justify-center border-2 border-black/40 w-1/4"><button className="w-full" type="button" buttonName="7" data-id="7" onClick={this.clickHandler.bind(this)}>7</button></li>
```
Comment (TedLivist) [inline src/components/calculator.jsx]: - [x] **[OPTIONAL]** Consider looking into this warning in the console about the `buttonName` attribute.

  ryruo

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ohyajk,

Good job so far!
✔️ Use logic file
✔️ Use of props
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.jsx (hunk 2 above)
Comment (ohyajk) [reply src/components/calculator.jsx]: Thank you @TedLivist for your helpful review.

File: src/components/calculator.jsx (hunk 1 above)
Comment (ohyajk) [reply src/components/calculator.jsx:4]: Thank you @TedLivist for the helpful comment. I have updated my code, and it is showing clicked number as expected.

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @ohyajk,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**